examples/autosharding/cluster-role.yaml

```diff
@@ -23,6 +23,7 @@ rules:
   - namespaces
   - endpoints
   verbs:
+  - get
```

Comment:
Apologies ahead of time for a stupid question but:
1. Why do we need secrets in resources?
2. Won't adding "get" mean this service can read secrets across the entire cluster?

Reply:
There's already a `list` and a `watch` on secrets, so a `get` does not really change anything wrt what KSM can read from the cluster. However, issuing a `get` request for each resource can easily overwhelm the apiserver which is the reason why we want to rely on watches.

Reply:
1. kube-state-metrics exposes metrics about secrets so it needs rbac permissions to list and watch.
2. as I mentioned in https://github.com/kubernetes/kube-state-metrics/pull/1614#issuecomment-945730358, I don't think we should add "get" permissions here, but the default kube-state-metrics config already has access to secrets across the entire cluster since it can `list` and `watch` them.
That said, if you want to prevent it from gaining access to secrets, you can always do so by preventing kube-state-metrics from collecting metrics about secrets by removing secrets from the list of resources with the `--resources` flag and adjust the RBAC permissions.

Reply:
Fair enough and thanks for the advice. Perhaps I am just paranoid but I would never want to grant full read to secrets to any service. In my opinion this should be turned off by default.

Reply:
> In my opinion this should be turned off by default.

I wouldn't agree with that. Kubernetes secrets are not meant to store sensitive data in the first place so giving the permission to read them to a service should be completely normal.

Kubernetes also doesn't secure secrets by default, if you are storing sensitive data in them, then you most likely want to enable encryption at rest which is not something enabled by default.

In the same way, if you are not confident enough in the separation between the different tenants of your cluster, you might also want to remove the permission to read secrets from kube-state-metrics as it may be used to leak sensitive data.